src/main/java/hudson/plugins/spotinst/common/GroupLockingManager.java

```diff
@@ -0,0 +1,347 @@
+package hudson.plugins.spotinst.common;
+
+import hudson.plugins.spotinst.api.infra.ApiResponse;
+import hudson.plugins.spotinst.repos.ILockRepo;
+import hudson.plugins.spotinst.repos.RepoManager;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Calendar;
+import java.util.Date;
+
+public class GroupLockingManager {
+    //region constants
+    public static final Integer LOCK_TIME_TO_LIVE_IN_SECONDS = 60 * 3;
+
+    private static final String  GROUP_CANNOT_BE_CONNECTED_DESCRIPTION_FORMAT    =
+            "group '%s' cannot be connected - check cloud's configuration";
+    private static final String  LOCK_OK_STATUS                                  = "OK";
+    private static final Integer MILI_TO_SECONDS                                 = 1000;
+    private static final Integer SUSPENDED_GROUP_FETCHING_TIME_TO_LIVE_IN_MILLIS =
+            MILI_TO_SECONDS * LOCK_TIME_TO_LIVE_IN_SECONDS + 10;
```

Comment:
in class GroupLockingManager , the + 10 should be second not millisecond, anyway it been used only in INITIALIZING_PERIOD and there you return to second, i think you can use only : INITIALIZING_PERIOD_SECOND = LOCK_TIME_TO_LIVE_IN_SECONDS + 10  , maybe consider use KEEP_ALIVE_MAXIMUM_ASSUMED_RUNNING_TIME = 10

Reply:
done